### PR TITLE
Extended bug fix in commit 3294b3 to SimpleJsonReader(Stream) constructor

### PR DESCRIPTION
**Pull request by maschram**

### CODE REVIEW

**Comment by chwarr** [comment]
The fix looks fine to me, Can you also add a test case or two that would fail without your fix but pass with it?

**Comment by maschram** [comment]
Added unit test

File: cs/test/core/JsonParsingTests.cs [hunk 1]
```diff
@@ -703,5 +703,25 @@ static T ParseJson<T>(string json) where T : new()
...
         }
+
+        [Test]
+        public void JsonParsing_DateAsStringInStreamConstructor()
+        {
+            string timestamp = "2017-03-07T04:47:50.1145227Z";
+            string json = $"{{\"timestamp\":\"{timestamp}\"}}";
+
+            DateAsString target;
+            using (MemoryStream ms = new MemoryStream())
```
Comment (chwarr) [inline cs/test/core/JsonParsingTests.cs]: How about the following instead?

```
var ms = new MemoryStream(Encoding.UTF8.GetBytes(json));
var reader = new SimpleJsonReader(ms);
var target = Deserialize<DataAsString>.From(reader);
Assert.AreEqual(timestamp, target.timestamp);
```

We can also skip disposing the MemoryStream. MemoryStream.Dispose doesn't do anything interesting.

Comment (chwarr) [inline cs/test/core/JsonParsingTests.cs]: The Travis CI build failed for as the `reader` constructed here never gets used. My proposed change will fix this problem as well.

**Comment by chwarr** [comment]
LGTM. I'll merge after the CI builds pass.

**Pull request closed without merging**